mitmproxy/addons/proxyauth.py

```diff
@@ -99,6 +101,20 @@ def check(self, f: http.HTTPFlow) -> Optional[Tuple[str, str]]:
         elif self.htpasswd:
             if self.htpasswd.check_password(username, password):
                 return username, password
+        elif self.ldapserver:
+            if not username or not password:
+                return None
+            dn = ctx.options.proxyauth.split(":")[2]
+            parts = dn.split("?")
+            conn = ldap3.Connection(
```

Comment:
Ok. We should add a comment that this is a potential performance bottleneck, because we are doing these connections in the main thread where they block everything else. If this becomes an actual issue, we can think about adding `@concurrent` to the respective addon hooks.